packages/wallet/src/KeyManagement/types.ts

```diff
@@ -72,7 +89,7 @@ export type GetPassword = (noCache?: true) => Promise<Uint8Array>;
 export interface KeyAgent {
```

Comment:
```suggestion
export interface KeyAgent<SerializableKeyAgentData> {
```
Re: https://github.com/input-output-hk/cardano-js-sdk/pull/189/files#r825305987
How about keeping the implementation concerns out of this scope, and use a generic type instead?

Reply:
I've implemented it like that originally, but it just makes it more complicated to use in the end. The goal here is for the user (e.g. the wallet) be able to use the key agent without knowing the specific type of it (and it's serializable data). [restoreKeyAgent](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/wallet/src/KeyManagement/restoreKeyAgent.ts#L32) is a utility that abstracts the specifics of serializable data and wallet initialization. It makes it possible for the user to work with just a `KeyAgent`.